deploy/determined_deploy/aws/aws.py

```diff
@@ -15,11 +15,7 @@ def get_user(boto3_session: boto3.session.Session) -> str:
 
 def delete(stack_name: str, boto3_session: boto3.session.Session) -> None:
     ec2 = boto3_session.client("ec2")
```

Comment:
Can you leave a comment at the top of this function that explains why the ordering we have here is necessary?  The fact that it was a bug for so long is an indicator that it's not intuitively obvious.